### PR TITLE
State revert uses the build planner

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2226" title="DX-2226" target="_blank">StoryDX-2226</a>  `state revert --to` uses the BuildPlanner
  </td></table>
  <br />

### CODE REVIEW

**Review by mitchell-as — CHANGES_REQUESTED** [review]
Overall this looks great. A few things to tweak though, I think.

File: internal/locale/locales/en-us.yaml [hunk 1]
```diff
@@ -1703,7 +1703,7 @@ err_user_network_solution:
 err_revert_get_commit:
   other: "Could not fetch commit details for commit with ID: {{.V0}}"
 err_revert_commit:
-  other: "Could not revert{{.V0}} commit: {{.V1}}"
+  other: "Could not revert {{.V0}} commit: {{.V1}}"
```
[the remark this answers is not in the document]
Comment (MDrakos) [reply internal/locale/locales/en-us.yaml:1706]: Ahh okay, thanks for catching that!

File: test/integration/revert_int_test.go [hunk 2]
```diff
@@ -115,7 +115,6 @@ func (suite *RevertIntegrationTestSuite) TestRevertTo() {
 		e2e.OptArgs("history"),
 		e2e.OptWD(wd),
 	)
-	cp.Expect("Reverting to commit " + commitID)
```
[the remark this answers is not in the document]
Comment (MDrakos) [reply test/integration/revert_int_test.go:118]: Good catch, this is actually the commit message that is now different with the buildplanner

File: pkg/platform/api/buildplanner/model/buildplan.go [hunk 3]
```diff
@@ -408,6 +415,18 @@ type CreateProjectResult struct {
 	ProjectCreated *projectCreated `json:"createProject"`
 }
 
+type RevertedCommit struct {
```
[the remark this answers is not in the document]
Comment (MDrakos) [reply pkg/platform/api/buildplanner/model/buildplan.go]: Nope, it can be private

File: test/integration/revert_int_test.go [hunk 4]
```diff
@@ -115,10 +115,12 @@ func (suite *RevertIntegrationTestSuite) TestRevertTo() {
 		e2e.OptArgs("history"),
 		e2e.OptWD(wd),
 	)
+	cp.Expect("Revert to commit " + commitID)
 	cp.Expect("- argparse") // effectively reverting previous commit
 	cp.Expect("+ argparse") // commit being effectively reverted
 	cp.Expect("+ urllib3")  // commit reverted to
 	cp.Expect("+ python")   // initial commit
+	fmt.Println("Output: ", cp.Snapshot())
```
Comment (mitchell-as) [inline test/integration/revert_int_test.go:123]: I think this was accidentally left here instead of removed.

**Review by mitchell-as — CHANGES_REQUESTED** [review]

File: test/integration/revert_int_test.go (hunk 4 above)
Comment (MDrakos) [reply test/integration/revert_int_test.go]: Whoops, thanks for catching that!

**Review by mitchell-as — APPROVED** [review]

**Pull request merged**